src/libs/actions/Report.js

```diff
@@ -425,6 +417,56 @@ function removeOptimisticActions(reportID) {
     });
 }
 
+/**
+ * Fetch the iouReport and persist the data to Onyx.
+ *
+ * @param {Number} iouReportID - ID of the report we are fetching
+ * @param {Number} chatReportID - associated chatReportID, set as an iouReport field
+ * @returns {Promise}
```

Comment:
I'm definitely not a fan of this promise being returned, and I think there are several other patterns that we have which can remove the need to return the promise here. For sake of #urgency, I am fine merging this as-is and then cleaning it up later. Here are some of the solutions that I think would ultimately make the code cleaner (in order of preference).

1. Solve more of the issues in the API layer rather than forcing the front-end to do a lot of work to keep all the data happy. For example, rather than expecting the front-end to try and keep the reportIDs in sync, the server should push data to e.cash that is just put straight into Onyx using pusher (this was the original dream).
2. Rather than having two methods (`fetchIOUReportByID` and `fetchIOUReportByIDAndUpdateChatReport`), add a third parameter to this method called `shouldUpdateChatReport` and move the logic into this method.
3. Have a callback connected to Onyx which looks for changes to the IOU report (eg. like after it's fetched from the server) and then it does the appropriate actions to keep the reportIDs in sync. This is similar to the pattern that we have for fetching new report data ([here](https://github.com/Expensify/Expensify.cash/blob/3291bc239c3d2415cd5bd913df816a36fb1b29a2/src/libs/actions/Report.js#L1061-L1077)).

Reply:
Fair points. The original PR used suggestion #2, but this led to other issues as the logic became quite complex and I switched to two functions in an attempt to make the code simpler to understand. 

I am already planning to do #1, but I have added #3 to the [post-launch improvements ](https://github.com/Expensify/Expensify/issues/148974)on the IOU tracker as this seems a worthy improvement.

Thanks for the review!